Code/Framework/AzToolsFramework/Tests/Input/QtEventToAzInputMapperTests.cpp

```diff
@@ -512,4 +516,145 @@ namespace UnitTest
             return info.param.m_az.GetName();
         }
     );
+
+    struct MouseMoveParam
+    {
+        AzToolsFramework::CursorInputMode mode;
+        int iterations;
+        QPoint startPos;
+        QPoint deltaPos;
+        QPoint expectedPos;
+        const char* name;
+    };
+
+    class MoveMoveWrapParamQtEventToAzInputMapperFixture
+        : public QtEventToAzInputMapperFixture
+        , public ::testing::WithParamInterface<MouseMoveParam>
+    {
+    };
+
+    TEST_P(MoveMoveWrapParamQtEventToAzInputMapperFixture, MouseMove_NoAzHandlers_VerifyMouseMovmentViewport) 
+    {
+        // setup
+        const MouseMoveParam mouseMoveParam = GetParam();
+
+        AzFramework::InputChannelNotificationBus::Handler::BusConnect();
+        m_captureAzEvents = false;
+        m_inputChannelMapper->SetCursorMode(mouseMoveParam.mode);
+        m_rootWidget->move(100, 100);
+
+
+        MouseMove(m_rootWidget.get(), mouseMoveParam.startPos, QPoint(0,0));
+        for(float i = 0; i < mouseMoveParam.iterations; i++) {
+            MouseMove(m_rootWidget.get(), m_rootWidget->mapFromGlobal(QCursor::pos()), (mouseMoveParam.deltaPos / mouseMoveParam.iterations));
+        }
+
+        QPointF endPosition = m_rootWidget->mapFromGlobal(QCursor::pos());
+        EXPECT_NEAR(endPosition.x(), mouseMoveParam.expectedPos.x(), 1.0f);
+        EXPECT_NEAR(endPosition.y(), mouseMoveParam.expectedPos.y(), 1.0f);
+
+        // cleanup
+        m_rootWidget->move(0, 0);
+        m_inputChannelMapper->SetCursorMode(AzToolsFramework::CursorInputMode::CursorModeNone);
+        AzFramework::InputChannelNotificationBus::Handler::BusDisconnect();
+    }
+
+    INSTANTIATE_TEST_CASE_P(All, MoveMoveWrapParamQtEventToAzInputMapperFixture,
+        testing::Values(
```

Comment:
The parameterized test looks great! Excellent idea 👍 Do these all pass now? A sprinkling of a few additional comments might be nice to help the uninitiated decipher them but otherwise it looks great!

The last thing I'd recommend is one or two using the old 'clamped' mode just to ensure the cursor _does not_ wrap in that case and stays on the correct side of the screen

Reply:
probably also want to verify the accumulated value.

Reply:
Yes good shout :+1: